core/io.go

```diff
@@ -0,0 +1,98 @@
+package core
+
+import (
+	"bytes"
+	"errors"
+	"io"
+
+	"github.com/dicedb/dice/config"
+)
+
+type RESPParser struct {
+	c   io.ReadWriter
+	buf *bytes.Buffer
+}
+
+func NewRESPParser(c io.ReadWriter) *RESPParser {
+	return NewRESPParserWithBytes(c, []byte{})
+}
+
+func NewRESPParserWithBytes(c io.ReadWriter, initBytes []byte) *RESPParser {
+	var b []byte
+	var buf *bytes.Buffer = bytes.NewBuffer(b)
+	buf.Write(initBytes)
+	return &RESPParser{
+		c:   c,
+		buf: buf,
+	}
+}
+
+func (rp *RESPParser) DecodeOne() (interface{}, error) {
+	// assigning temporary buffer to read 512 bytes in one shot
+	// and reading them in a loop until we have all the data
+	// we want.
+	// note: the size 512 is arbitrarily chosen, and we can put
+	// a decent thought into deciding the optimal value (in case it affects the perf)
+	var buf []byte = make([]byte, config.IOBufferLength)
+
+	// keep reading the bytes from the buffer until the first /r is found
+	// note: there may be extra bytes read over the socket post /r
+	// but that is fine.
+	for {
+		n, err := rp.c.Read(buf)
+		if n <= 0 {
+			break
+		}
+		if err != nil {
+			if err == io.EOF {
+				break
+			}
+			return nil, err
+		}
+		rp.buf.Write(buf[:n])
+
+		if bytes.Contains(buf, []byte{'\r', '\n'}) {
+			break
+		}
+	}
```

Comment:
This is a security bug. This is no less then Cross Protocol Scripting.  
 
From client docs - https://redis.io/docs/reference/clients/

> Every client is also subject to a query buffer limit. This is a non-configurable hard limit that will close the connection when the client query buffer (that is the buffer we use to accumulate commands from the client) reaches 1 GB, and is actually only an extreme limit to avoid a server crash in case of client or server software bugs.